test/reply-error.test.js

```diff
@@ -336,3 +292,29 @@ test('should set the status code and the headers from the error object (from cus
     })
   })
 })
+
+test('Should throw when non-error value is used to reject a promise', t => {
+  t.plan(2)
+
+  // Tap patched this event and we have no chance to listen on it.
+  process.removeAllListeners('unhandledRejection')
```

Comment:
I fear the current listeners to `unhandledRejection` should be restored after this test ends.

Reply:
Yes, or I use `prependOnceListener`

Reply:
Fixed